request.el

```diff
@@ -1177,7 +1161,7 @@ See \"set-cookie-av\" in http://www.ietf.org/rfc/rfc2965.txt")
 START-URL is the URL requested."
   (cl-loop for prev-url = start-url then url
            for url in redirects
-           unless (and url (string-match url-nonrelative-link url))
+           unless (string-match url-nonrelative-link url)
```

Comment:
I'd like to remove this, but with your patch and without my `(and url ...)` hack, I get:
```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  string-match("\\`\\([a-zA-Z][-a-zA-Z0-9+.]*:\\)" nil)
  (if (string-match url-nonrelative-link url) (progn) (setq url (url-expand-file-name url prev-url)))
  (while (progn (setq prev-url (if --cl-var-- start-url url)) (consp --cl-var--)) (setq url (car --cl-var--)) (if (string-match url-nonrelative-link url) (progn) (setq url (url-expand-file-name url prev-url))) (setq --cl-var-- (cons url --cl-var--)) (setq --cl-var-- (cdr --cl-var--)) (setq --cl-var-- nil))
  (let* ((prev-url nil) (--cl-var-- redirects) (url nil) (--cl-var-- nil) (--cl-var-- t)) (while (progn (setq prev-url (if --cl-var-- start-url url)) (consp --cl-var--)) (setq url (car --cl-var--)) (if (string-match url-nonrelative-link url) (progn) (setq url (url-expand-file-name url prev-url))) (setq --cl-var-- (cons url --cl-var--)) (setq --cl-var-- (cdr --cl-var--)) (setq --cl-var-- nil)) (nreverse --cl-var--))
  request--curl-absolutify-redirects("http://builder:8002/manage/v2" (nil))
  (let* ((--cl-var-- (request--curl-absolutify-redirects start-url (mapcar (function (lambda (response) (request-response-header response "location"))) history))) (url nil) (--cl-var-- (cdr history)) (response nil)) (while (and (consp --cl-var--) (progn (setq url (car --cl-var--)) (consp --cl-var--))) (setq response (car --cl-var--)) (progn (or (and (memq (type-of response) cl-struct-request-response-tags) t) (signal (quote wrong-type-argument) (list (quote request-response) response))) (let* ((v response)) (aset v 6 url))) (setq --cl-var-- (cdr --cl-var--)) (setq --cl-var-- (cdr --cl-var--))) nil)
```

Reply:
Ah, yes.  I put in a null check upstream.